### PR TITLE
[Docs] Replace markdownlint with mdformat for avoiding installing ruby

### PR DESCRIPTION
**Pull request by zhouzaida**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Replace markdownlint with mdformat to avoid installing ruby

## Modification

docs/*.md

## BC-breaking (Optional)

No

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,16 +25,19 @@ repos:
...
   - repo: https://github.com/codespell-project/codespell
     rev: v2.1.0
     hooks:
       - id: codespell
+  - repo: https://github.com/executablebooks/mdformat
+    rev: 0.7.14
+    hooks:
+      - id: mdformat
+        args: ["--number"]
+        additional_dependencies:
```
Comment (ZwwWayne) [inline .pre-commit-config.yaml:37]: some rules are skipped in the original markdownlint, should we also set some rules here to lose some restriction?

File: docs/en/get_started/build.md [hunk 2]
```diff
@@ -74,41 +75,41 @@ You should know how to set up environment variables, especially `Path`, on Windo
 
 1. Launch Anaconda prompt from Windows Start menu
 
-    Do not use raw `cmd.exe` s instruction is based on PowerShell syntax.
+   Do not use raw `cmd.exe` s instruction is based on PowerShell syntax.
```
Comment (ZwwWayne) [inline docs/en/get_started/build.md:78]: we may check whether the indentation is expected after modification

File: docs/zh_cn/deployment/tensorrt_custom_ops.md [hunk 3]
```diff
@@ -2,61 +2,61 @@
 
 <!-- TOC -->
 
-- [TensorRT自定义算子](#tensorrt自定义算子)
+- [TensorRT自定义算子](#tensorrt%E8%87%AA%E5%AE%9A%E4%B9%89%E7%AE%97%E5%AD%90)
```
Comment (ZwwWayne) [inline docs/zh_cn/deployment/tensorrt_custom_ops.md:5]: this modification is a bit wierd, makes it hard to write Chinese doc.

File: docs/en/get_started/build.md (hunk 2 above)
Comment (zhouzaida) [reply docs/en/get_started/build.md:78]: It works as expected.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (zhouzaida) [reply .pre-commit-config.yaml:37]: `mdformat` does not accept those rules to skip checking. Maybe we need to discuss those rules.

File: docs/zh_cn/deployment/tensorrt_custom_ops.md (hunk 3 above)
Comment (zhouzaida) [reply docs/zh_cn/deployment/tensorrt_custom_ops.md:5]: ugly indeed

Comment (zhouzaida) [reply docs/zh_cn/deployment/tensorrt_custom_ops.md:5]: I opened an issue at mdformat. https://github.com/executablebooks/mdformat/issues/322

File: .pre-commit-config.yaml (hunk 1 above)
Comment (zhouzaida) [reply .pre-commit-config.yaml:37]: https://github.com/executablebooks/mdformat/issues/53

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -25,16 +25,19 @@ repos:
...
+          - mdformat-gfm
```
Comment (zhouzaida) [inline .pre-commit-config.yaml:38]: We can remove this dependency if there is no need to format tables.

Comment (ZwwWayne) [reply .pre-commit-config.yaml:38]: we can keep this since it is done automatically.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**